src/infrastructure/database/core.py

```diff
@@ -7,17 +9,31 @@
 
 from config import settings
 
-__all__ = ["get_session", "engine"]
-
+__all__ = ["engine", "session_manager"]
 
 engine = create_async_engine(
-    settings.database.url, future=True, pool_pre_ping=True, echo=False
+    settings.database.url,
+    future=True,
+    pool_pre_ping=True,
+    echo=False,
+    pool_size=settings.database.pool_size,
 )
 
 
-def get_session() -> AsyncSession:
-    async_session_factory = sessionmaker(engine, class_=AsyncSession)
-    AsyncScopedSession = async_scoped_session(
-        async_session_factory, scopefunc=lambda: None
-    )
-    return AsyncScopedSession()
+class SessionManager:
+    def __init__(self):
+        self.session = None
+
+    def get_session(self):
+        if not self.session:
+            async_session_factory = sessionmaker(
+                engine, class_=AsyncSession, expire_on_commit=False
+            )
+            AsyncScopedSession = async_scoped_session(
+                async_session_factory, scopefunc=asyncio.current_task
+            )
+            self.session = AsyncScopedSession()
+        return self.session
+
+
+session_manager = SessionManager()
```

Comment:
If this class is created only for tests, could we get it back and mock the session inside `BaseCRUD.__ini__.session` instead?

Reply:
Yea :)
So check in which placed do you use it in general in the whole project

Reply:
Yea :)
So check in which placed do you use it in general in the whole project

Reply:
I will add additional decorator for transaction(if success commit else rollback), now it is used only in rollback decorator